### PR TITLE
implemented solo_variatation, zoomout, expand features

### PR DESCRIPTION
**Pull request by tangramor**
实现了以下功能：

- [x]  `/v1/api/trigger/solo_variation`：Make Variations
- [x]  `/v1/api/trigger/zoomout`：Zoom Out 2x/1.5x
- [x]  `/v1/api/trigger/expand`：⬅️ ➡️ ⬆️ ⬇️

### CODE REVIEW

**Comment by yokonsan** [comment]
Thank you, pr.
I've been a little busy at work lately, so I'll merge later.

**Comment by CJReinforce** [comment]
> 实现了以下功能：
> 
> * [x]   `/v1/api/trigger/solo_variation`：Make Variations
> * [x]   `/v1/api/trigger/zoomout`：Zoom Out 2x/1.5x
> * [x]   `/v1/api/trigger/expand`：⬅️ ➡️ ⬆️ ⬇️

Thanks for your work! There may be two minor problems:

1. `/solo_variation` seems does not work. 
After `curl`, I have got the response `{"trigger_id":"xxxxxxx","trigger_type":"solo_variation"}`, and the log of server is shown as following, but then nothing happened anymore.
```log
2023-08-02 18:07:44 | DEBUG | 807531-281472936744784 | "/root/program/forked-midjourney-api/util/_queue.py:61":_exec - Task[xxxxxxxx] start execution: solo_variation((), {'index': 1, 'msg_id': 'xxxxxxxx', 'msg_hash': 'xxxxxxxx', 'trigger_id': 'xxxxxxxx'})
INFO:     206.190.237.43:52344 - "POST /v1/api/trigger/solo_variation HTTP/1.1" 200 OK
2023-08-02 18:07:44 | DEBUG | 807531-281472936744784 | "/root/program/forked-midjourney-api/util/fetch.py:47":fetch - Fetch: https://discord.com/api/v9/interactions, {'data': '{"type": 3, "application_id": "xxxxxxxx", "guild_id": "xxxxxxxx", "channel_id": "xxxxxxxx", "session_id": "xxxxxxxx", "data": {"component_type": 2, "custom_id": "MJ::JOB::variation::1::xxxxxxxx::SOLO"}, "message_flags": 0, "message_id": "xxxxxxxx"}'}
```
4. Suggest to change the code `@router.post("/solo_variation")` to `@router.post("/solo_variation", response_model=TriggerResponse)` (the same to `/expand` and `/zoomout`). The latter can add some meta info like `{"message": True}` in the response.

File: lib/api/discord.py [hunk 1]
```diff
@@ -144,6 +147,42 @@ async def variation(index: int, msg_id: str, msg_hash: str, **kwargs):
...
 
 
+async def solo_variation(index: int, msg_id: str, msg_hash: str, **kwargs):
+    kwargs = {
+        "message_flags": 0,
+        "message_id": msg_id,
+    }
+    payload = _trigger_payload(3, {
+        "component_type": 2,
+        "custom_id": f"MJ::JOB::variation::{index}::{msg_hash}::SOLO"
```
Comment (CJReinforce) [inline lib/api/discord.py:157]: solo variation has 2 modes: Vary(strong) and Vary(Subtle). The `custom_id` should be `f"MJ::JOB::high_variation::{index}::{msg_hash}::SOLO"` and `f"MJ::JOB::low_variation::{index}::{msg_hash}::SOLO"` respectively. That's why `/solo_variation` doesn't work now.

Comment (tangramor) [reply lib/api/discord.py]: Thanks for the test, I tested it in my env, it works...
And you can find the official parameters here:
image

Comment (CJReinforce) [reply lib/api/discord.py]: image
Thanks for reply! But our buttons look different, I don't have the "Make Variation" button, instead, there are "Vary(Strong)" and "Vary(Subtle)" options.

Comment (tangramor) [reply lib/api/discord.py]: OK, I added Vary stuble/strong features for your condition.
It may caused by different user types. I am using a pro user...

Comment (CJReinforce) [reply lib/api/discord.py]: Thanks so much! :relaxed: Right, it‘s due to different user types. I just bought the basic plan.  Thank you again.

**Pull request merged**